src/PAModel/Serializers/SourceSerializer.cs

```diff
@@ -411,9 +427,25 @@ private static void LoadSourceFiles(CanvasDocument app, DirectoryReader director
 
             foreach (var file in directory.EnumerateFiles(CodeDir, "*.fx.yaml", searchSubdirectories: false))
             {
-                AddControl(app, file._relativeName, false, file.GetContents(), errors);
+                // Create new fxYamlGroup object
+                fxYamlGroup currFxYamlGroup;
+
+                // Set it's values
+                currFxYamlGroup.file = file;
+                currFxYamlGroup.canSearchSubdirs = false;
+
+                // Add fxYamlGroup object to the list
+                fxYamlList.Add(currFxYamlGroup);
             }
 
+            // Sort the list of fxYaml files alphabetically
+            fxYamlList.OrderBy((fxYamlGroup) => fxYamlGroup.file._relativeName);
+
+            // Add control for every .fx.yaml file
+            fxYamlList.ForEach(delegate(fxYamlGroup group ) {
+            AddControl(app, group.file._relativeName, group.canSearchSubdirs, group.file.GetContents(), errors);
```

Comment:
I don't think this is actually solving the issue mentioned in the PR re the order of the controls within the file. This looks like it's sorting at the file level, which is not going to have any effect since the filesystem, and zip files are unordered.

Reply:
I want to order it by what the control name is going to be, but it doesn't seem like I have any way to do that unless I want to invoke the parser in this section of the code. I may have to move to another section of code unless it's okay to invoke the parser up here, just to grab the controlIR name and order alphabetically